### PR TITLE
Append Elastic/synthetics to UA string

### PR DESCRIPTION
**Pull request by shahzad31**
Fixes https://github.com/elastic/synthetics/issues/232

Appends Elastic/Synthetics to UA string. 

This PR uses a pretty ugly way to determin current user agent string and append the desired string into it.

Not sure if there is a way better way to do this. Struggling with playwright docs on how to get the current user agent string without opening a browser context.

### CODE REVIEW

**Review by vigneshshanmugam — CHANGES_REQUESTED** [review]
Even though this works, This would be the last approach we would need to do as it slows down the startup time of our runner. Please look in to - https://chromedevtools.github.io/devtools-protocol/tot/Browser/#method-getVersion

which can be used to get the version and can be used for setting the global user agent context. Let me know if you need any help.

**Comment by vigneshshanmugam** [comment]
We discussed offline and decided to go with using browser based session approach to get the user agent. 

Benchmarks if anyone is curious
```
Session: 3.505ms
Context: 245.072ms
```

**Comment by shahzad31** [comment]
i tested manualy, it seems to work with iframe and new window as well.

will try to write some tests for those use cases.

**Comment by shahzad31** [comment]
Added tests for iframe and popup

File: src/core/gatherer.ts [hunk 1]
```diff
@@ -46,12 +48,26 @@ export class Gatherer {
...
     }
-    const context = await Gatherer.browser.newContext(playwrightOptions);
+
+    const context = await Gatherer.browser.newContext({
+      ...playwrightOptions,
+      userAgent: await Gatherer.getUserAgent(),
+    });
+
     const page = await context.newPage();
+
```
Comment (vigneshshanmugam) [inline src/core/gatherer.ts:58]: nit: can we get rid of these empty spaces? both in tests and code? would like to keep it similar to existing code.

File: __tests__/core/gatherer.test.ts [hunk 2]
```diff
@@ -54,4 +63,84 @@ describe('Gatherer', () => {
...
+        ' Elastic/Synthetics'
+      );
+
+      expect(await page1.textContent('body')).toEqual('Not found');
+
+      await Gatherer.dispose(driver);
+      await Gatherer.stop();
+    });
+
+    it('works with iframe content', async () => {
```
Comment (vigneshshanmugam) [inline __tests__/core/gatherer.test.ts:122]: I would remove this iframe test in favor of request test. Or merge it on the previous popup test to look for request to `popup.html` and check if request.headers.userAgent contains the Elastic identifier. 

Iframes tests are unnecessary as they inherit the parent document.

File: __tests__/core/gatherer.test.ts [hunk 3]
```diff
@@ -26,10 +26,19 @@
...
+import { devices } from 'playwright-chromium';
+import { Server } from '../utils/server';
+import exp = require('constants');
 
 jest.mock('../../src/plugins/network');
 
 describe('Gatherer', () => {
+  let server: Server;
+
+  beforeAll(async () => {
```
Comment (vigneshshanmugam) [inline __tests__/core/gatherer.test.ts:38]: Close the server on afterAll.

File: __tests__/core/gatherer.test.ts [hunk 4]
```diff
@@ -26,10 +26,19 @@
 import { Gatherer } from '../../src/core/gatherer';
 import { PluginManager } from '../../src/plugins';
 import { wsEndpoint } from '../utils/test-config';
+import { devices } from 'playwright-chromium';
+import { Server } from '../utils/server';
+import exp = require('constants');
```
Comment (vigneshshanmugam) [inline __tests__/core/gatherer.test.ts:31]: bad import.

File: src/core/gatherer.ts [hunk 5]
```diff
@@ -27,7 +27,9 @@ import { chromium, ChromiumBrowser } from 'playwright-chromium';
 import { PluginManager } from '../plugins';
 import { RunOptions } from './runner';
 import { log } from './logger';
-import { Driver } from '../common_types';
+import { Driver, PlaywrightOptions } from '../common_types';
+import { Protocol } from 'playwright-chromium/types/protocol';
```
Comment (vigneshshanmugam) [inline src/core/gatherer.ts:31]: need to remove below two imports

File: __tests__/core/gatherer.test.ts [hunk 6]
```diff
@@ -54,4 +63,84 @@ describe('Gatherer', () => {
     expect(network.start).toHaveBeenCalled();
     await Gatherer.stop();
   });
+
+  describe('Append Elastic/Synthetics to UA', () => {
```
Comment (vigneshshanmugam) [inline __tests__/core/gatherer.test.ts:67]: nit: Elastic UA identifier

**Review by vigneshshanmugam — COMMENTED** [review]
Generally looks good, good to go after removing some bad import statements.

File: src/core/gatherer.ts (hunk 1 above)
Comment (shahzad31) [reply src/core/gatherer.ts:58]: i really like spaced out code :D

Comment (vigneshshanmugam) [reply src/core/gatherer.ts]: I see 😄

**Review by vigneshshanmugam — APPROVED** [review]
LGTM 🎉

**Pull request merged**